presto-hive/src/main/java/io/prestosql/plugin/hive/HiveWriterFactory.java

```diff
@@ -549,6 +549,37 @@ else if (insertExistingPartitionsBehavior == InsertExistingPartitionsBehavior.ER
                 hiveWriterStats);
     }
 
+    private String computeFileName(OptionalInt bucketNumber)
+    {
+        if (isCreateTransactionalTable) {
+            // Hive is very opinionated on naming pattern for "original" files of transactional tables.
+            // As currently CTAS for transactional tables in Presto creates non-transactional ("original") files.
+            // we use special file naming mode in this case.
+            //
+            // For bucketed tables we drop query id from file names and just leave <bucketId>_0
+            // For non bucketed tables we use 000000_<uuid_as_number>
```

Comment:
It seems like `bucketNumber.isPresent()` is the more important condition than `isTransactionalTable`.
So I'd structure the code different, with bucket being the top if.